datahub/dataset/company_future_interest_countries/test/test_views.py

```diff
@@ -1,39 +1,34 @@
-import factory
 import pytest
 from django.urls import reverse
 from rest_framework import status
 
-from datahub.company.models import Company
-from datahub.company.test.factories import CompanyFactory
-from datahub.core import constants
+from datahub.company.models import CompanyExportCountry
+from datahub.company.test.factories import CompanyExportCountryFactory, CompanyFactory
 from datahub.core.test.factories import to_many_field
 from datahub.dataset.core.test import BaseDatasetViewTest
 from datahub.metadata.models import Country
 
 
-class CompanyFutureInterestCountriesFactory(factory.django.DjangoModelFactory):
-    """Returns a company-future_interest_country relationship"""
-
-    company = factory.SubFactory(CompanyFactory)
-    country_id = constants.Country.united_kingdom.value.id
-
-    class Meta:
-        model = Company.future_interest_countries.through
-
-
 class CompanyWithFutureInterestCountriesFactory(CompanyFactory):
     """Overrides Company Factory to provide a future interest country"""
 
     @to_many_field
-    def future_interest_countries(self):
-        """Return stubbed countries"""
-        return list(Country.objects.all()[:1])
+    def export_countries(self):
+        """Return stubbed export countries"""
+        return [
+            CompanyExportCountryFactory(
+                status=CompanyExportCountry.Status.FUTURE_INTEREST,
+            ),
+        ]
 
 
 def get_expected_data_from_company(company):
-    """Returns company future interest countries data as a list of dictionaries"""
-    values = company.future_interest_countries.through.objects.filter(
-        company_id=company.id,
+    """
+    Returns company export_countries data with status `future_interest`
+    as a list of dictionaries
+    """
+    values = company.export_countries.all().filter(
```

Comment:
Could get rid of the `.all()` here:

```suggestion
    values = company.export_countries.filter(
```